File: src/playback_controller.py
from threading import Lock


class PBControl:
    def __init__(self) -> None:
        self._tmp_mult = 1.0
        self._lock = Lock()
        self._exp=90
    def set_tmp_mult(self, value: float) -> None:
        value = max(0.5, min(2.0, value))

        with self._lock:
            self._tmp_mult = value

    def get_tmp_mult(self) -> float:
        with self._lock:
            return self._tmp_mult
    def set_exp(self,value :int)->None:
        value = max(0, min(127, value))

        with self._lock:
            self._exp= value

    def get_exp(self) -> int:
        with self._lock:
            return self._exp

File: web/src/playbackcontroller.js
export class PBC{
    constructor() {
        this._tmpMult=1.0
        this._expression=90

    }


    reset(){
        this._tmpMult=1.0
        this._expression=0.9
    }

    setTempMult(value) {
        this._tmpMult=Math.max(0.5 , Math.min(2.0,value))
    }
    getTempMult(value){
        return this._tmpMult
    }
    setExp(value) {
        this._expression=Math.max(0,Math.min(127,Math.round(value)))
    }
    getExp(value) {
        return this._expression
    }


}


